python/lsst/daf/butler/script/pruneCollection.py

```diff
@@ -42,14 +64,89 @@ def pruneCollection(repo, collection, purge, unstore, unlink):
         Same as the ``unstore`` argument to ``Butler.pruneCollection``.
     unlink: `list` [`str`]
         Same as the ``unlink`` argument to ``Butler.pruneCollection``.
+    confirm : `bool`
+        If `True` will produce a table of collections that will be removed for
+        display to the user.
+
+    Returns
+    -------
+    collections : `astropy.table.Table`
+        The table containing collections that will be removed, their type, and
+        the number of datasets in the collection if applicable.
     """
-    butler = Butler(repo, writeable=True)
-    try:
-        butler.pruneCollection(collection, purge, unstore, unlink)
-    except PurgeWithoutUnstorePruneCollectionsError as e:
-        raise TypeError("Cannot pass --purge without --unstore.") from e
-    except RunWithoutPurgePruneCollectionsError as e:
-        raise TypeError(f"Cannot prune RUN collection {e.collectionType.name} without --purge.") from e
-    except PurgeUnsupportedPruneCollectionsError as e:
-        raise TypeError(
-            f"Cannot prune {e.collectionType} collection {e.collectionType.name} with --purge.") from e
+    class CollectionInfo:
+        """Lightweight container to hold the type of collection and the number
+        of datasets in the collection if applicable."""
+        def __init__(self, count: Union[None, int], type: Union[str]):
```

Comment:
You can write the first one as Optional[int] and why is the second one a union at all?